src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -14,45 +14,505 @@ IPhysicalDevice::IPhysicalDevice(core::smart_refctd_ptr<system::ISystem>&& s, co
+    if (m_features.rasterizationOrder) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_RASTERIZATION_ORDER");
+    if (m_features.shaderExplicitVertexParameter) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SHADER_EXPLICIT_VERTEX_PARAMETER");
+    if (m_features.shaderInfoAMD) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SHADER_INFO_AMD");
+    if (m_features.hostQueryReset) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_HOST_QUERY_RESET");
+    if (m_features.pipelineCreationCacheControl) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_PIPELINE_CREATION_CACHE_CONTROL");
+    if (m_features.colorWriteEnable) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_COLOR_WRITE_ENABLE");
+    if (m_features.conditionalRendering) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_CONDITIONAL_RENDERING");
+    if (m_features.inheritedConditionalRendering) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_INHERITED_CONDITIONAL_RENDERING");
+    if (m_features.deviceMemoryReport) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_DEVICE_MEMORY_REPORT");
+    if (m_features.fragmentDensityMap) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_FRAGMENT_DENSITY_MAP");
+    if (m_features.fragmentDensityMapDynamic) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_FRAGMENT_DENSITY_MAP_DYNAMIC");
+    if (m_features.fragmentDensityMapNonSubsampledImages) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_FRAGMENT_DENSITY_MAP_NON_SUBSAMPLED_IMAGES");
+    if (m_features.fragmentDensityMapDeferred) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_FRAGMENT_DENSITY_MAP_DEFERRED");
+    if (m_features.robustImageAccess) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_ROBUST_IMAGE_ACCESS");
+    if (m_features.inlineUniformBlock) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_INLINE_UNIFORM_BLOCK");
+    if (m_features.descriptorBindingInlineUniformBlockUpdateAfterBind) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_DESCRIPTOR_BINDING_INLINE_UNIFORM_BLOCK_UPDATE_AFTER_BIND");
+    if (m_features.rectangularLines) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_RECTANGULAR_LINES");
+    if (m_features.bresenhamLines) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_BRESENHAM_LINES");
+    if (m_features.smoothLines) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_SMOOTH_LINES");
+    if (m_features.stippledRectangularLines) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_STIPPLED_RECTANGULAR_LINES");
+    if (m_features.stippledBresenhamLines) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_STIPPLED_BRESENHAM_LINES");
+    if (m_features.stippledSmoothLines) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_STIPPLED_SMOOTH_LINES");
+    if (m_features.memoryPriority) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_MEMORY_PRIORITY");
+    if (m_features.robustBufferAccess2) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_ROBUST_BUFFER_ACCESS_2");
+    if (m_features.robustImageAccess2) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_ROBUST_IMAGE_ACCESS_2");
+    if (m_features.nullDescriptor) addGLSLDefineToPool(pool, "NBL_GLSL_FEATURE_NULL_DESCRIPTOR");
```

Comment:
@achalpandeyy @Erfan-Ahmadi interesting, does it do what I think it does?

Reply:
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkWriteDescriptorSet.html
> If the [nullDescriptor](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#features-nullDescriptor) feature is enabled, the buffer, acceleration structure, imageView, or bufferView can be [VK_NULL_HANDLE](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_NULL_HANDLE.html). Loads from a null descriptor return zero values and stores and atomics to a null descriptor are discarded. A null acceleration structure descriptor results in the miss shader being invoked.

Reply:
i dont think the descriptor sets initialize to nulls though

Reply:
@devshgraphicsprogramming You may think it's about the desriptorset binding thing but it's `nullDescriptor` not `nullDescriptorSet`, It basically allows you to have null descriptors (VK_NULL_HANDLE passed to vkWriteDescriptorSet structs) and then read return 0 and writes are discarded

Reply:
ok thanks!